### DISCUSSION TITLE
CellProfiler 4 EOL patch branch?

### DISCUSSION THREAD

**Discussion by bethac07** [CellProfiler/CellProfiler#4667] [General]
In CellProfiler versions 2 and 3, because either a) the next major version was taking a very long time to come out and/or b) we knew people were resistant to switching (this was more for 2 than 3, because 3->4 was both a much smaller UI change and involved the Python2 EOL we were basically like "too bad, switch anyway"), we ended up with "patch branches" wherein we cherry-picked over major bug fixes from the new major version candidates onto the last previous-major-version release. You can see examples of this in [the CellProfiler 2.3.1 tag](https://github.com/CellProfiler/CellProfiler/tree/v2.3.1) (which was 6 months after the release of 3.0.0 and only a week before the first 3.1 - tagged version) and the [3.1.9 branch](https://github.com/CellProfiler/CellProfiler/tree/v3.1.9). The goal is not to ever merge the branch, but merely allow for the creation of new releases that fix major non-breaking-change bugs without having to wait for the large engineering projects we're trying to solve in the next major release. 

I don't really LIKE doing this, because it can get messy (see [this comment](https://github.com/CellProfiler/CellProfiler/issues/4442#issuecomment-1062238910), detailing a bug that crept into 4 when functionality that was built into the 3.1.9 patch release was then poorly ported by me into CP4 - though at least here the versions are far less divergent since we haven't language-swapped), but if we think there is a "showstopper" bug and that the next major version release is a long ways away, we've sometimes decided it is the least-worst solution. #4666 is the first since 4.2.4 release (which granted was not that long ago) that I think would potentially qualify, since it seems that all loading of 3D label matrices is broken, and while that's not our number one use case, given increasing popularity of things like Cellpose and StarDist it's probably one we don't want to break support of entirely. 

What do we think? Should we start a post-4.2.4 branch? Or do we just tell folks to hold on?

**Comment by rsenft1** [comment]
I'd vote for yes, especially since it seems like #4666 is a relatively simple fix. Adding that and a new test or two seems very useful for a more feature complete/non-busted final v4.
